Source/WebKit/UIProcess/API/glib/WebKitDownload.cpp

```diff
@@ -449,52 +456,73 @@ WebKitURIRequest* webkit_download_get_request(WebKitDownload* download)
  * webkit_download_get_destination:
  * @download: a #WebKitDownload
  *
- * Obtains the URI to which the downloaded file will be written.
+ * Obtains the destination to which the downloaded file will be written.
  *
- * You
- * can connect to #WebKitDownload::created-destination to make
+ * You can connect to #WebKitDownload::created-destination to make
  * sure this method returns a valid destination.
  *
- * Returns: the destination URI or %NULL
+ * Returns: (nullable): the destination or %NULL
  */
 const gchar* webkit_download_get_destination(WebKitDownload* download)
 {
-    g_return_val_if_fail(WEBKIT_IS_DOWNLOAD(download), 0);
+    g_return_val_if_fail(WEBKIT_IS_DOWNLOAD(download), nullptr);
 
-    return download->priv->destinationURI.data();
+#if !ENABLE(2022_GLIB_API)
+    if (download->priv->destinationURI)
+        return download->priv->destinationURI.get();
+#endif
+    return download->priv->destination.get();
 }
 
 /**
  * webkit_download_set_destination:
  * @download: a #WebKitDownload
- * @uri: the destination URI
+ * @destination: the destination
```

Comment:
I think this is worth trying because it's nice to avoid duplicating the comments, but it is an introspection API break, so in theory the rename could break apps that work today. E.g. any app that does `download.set_destination(uri='file:///foo')` will break. Unfortunate. But we got away with doing this once in the past for blacklist/whitelist -> denylist/allowlist, so... hopefully no apps are doing this? It would be a little weird to specify the name of an argument when calling a function with only one argument.

So up to you. I think it's worth trying, but it is an API break.

Reply:
Actually there is another introspection API break here anyway (added nullable annotation to get_destination) and that one is really needed for correctness, so might as well try this too. _Shrug_.

Reply:
I don't think this will break anything.